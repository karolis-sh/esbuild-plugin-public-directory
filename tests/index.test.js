const os = require('os');
const path = require('path');
const { build } = require('esbuild');
const dirTree = require('directory-tree');

const publicDir = require('../lib');

const getResult = (dir) => {
  const result = [];
  dirTree(dir, null, (item, pathname, stats) => {
    result.push({ file: path.relative(dir, pathname), size: stats.size });
  });
  return result;
};

const TEMP_DIR = path.join(os.tmpdir(), `esbuild-plugin-public-directory/${Date.now()}`);

it('should use outdir option', async () => {
  const dir = path.join(TEMP_DIR, 'outdir-1');

  await build({
    entryPoints: [path.join(__dirname, 'src/index.js')],
    outdir: dir,
    plugins: [publicDir({ entry: path.join(__dirname, 'public') })],
  });

  expect(getResult(dir)).toMatchSnapshot();
});

it('should respect absWorkingDir option', async () => {
  const dir = path.join(TEMP_DIR, 'absWorkingDir-1');

  await build({
    entryPoints: ['src/index.js'],
    outdir: dir,
    plugins: [publicDir()],
    absWorkingDir: __dirname,
  });

  expect(getResult(dir)).toMatchSnapshot();
});

it('should use outfile option', async () => {
  const dir = path.join(TEMP_DIR, 'outfile-1');

  await build({
    entryPoints: [path.join(__dirname, 'src/index.js')],
    outfile: path.join(dir, 'specificied/sample.js'),
    plugins: [publicDir()],
    absWorkingDir: __dirname,
  });

  expect(getResult(dir)).toMatchSnapshot();
});
