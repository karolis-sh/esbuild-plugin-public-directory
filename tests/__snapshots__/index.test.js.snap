// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should respect absWorkingDir option 1`] = `
[
  {
    "file": "data/test.json",
    "size": 19,
  },
  {
    "file": "index.js",
    "size": 30,
  },
  {
    "file": "robots.txt",
    "size": 4,
  },
]
`;

exports[`should use outdir option 1`] = `
[
  {
    "file": "data/test.json",
    "size": 19,
  },
  {
    "file": "index.js",
    "size": 30,
  },
  {
    "file": "robots.txt",
    "size": 4,
  },
]
`;

exports[`should use outfile option 1`] = `
[
  {
    "file": "specificied/data/test.json",
    "size": 19,
  },
  {
    "file": "specificied/robots.txt",
    "size": 4,
  },
  {
    "file": "specificied/sample.js",
    "size": 30,
  },
]
`;
